File: c/1003.c

#include <stdio.h>
#include <string.h>


int main(){
    int n;
    int i,j;
    scanf("%d",&n);
    for (j=0; j<n;j++) {
        char str[100]={NULL};
        scanf("%s",str);
        int count_P=0;
        int count_T=0;
        int jump=0;
        int sign=0;
        int sign_P=0;
        int sign_T=0;
        int sign_end=0;
        int num_a;
        int num_end;
        for (i=0; i<100; i++) {
            if (str[i]!='P'&&str[i]!='A'&&str[i]!='T') {
                printf("NO\n");
                jump=1;
                break;
            }

            if (str[i]=='P') {
                count_P++;
                sign_P=i;
                if (sign==0) {
                    sign=1;
                }
            }
            
            if (str[i]=='T') {
                if (sign==0) {
                    printf("NO\n");
                    jump=1;
                    break;
                }
                if (i==sign_P+1) {
                    printf("NO\n");
                    jump=1;
                    break;
                }
                count_T++;
                sign_T=i;
            }
            if (str[i+1]=='\0') {
                sign_end=i;
                break;
            }
        }
        if (jump==0) {
            num_a=sign_T-sign_P-2;
            num_end=sign_end-sign_T;
            if (count_P==1&&count_T==1) {
                if (num_end>=(num_a+1)*sign_P) {
                    printf("YES\n");
                }
                else{
                    printf("NO\n");
                }
            }
            else{
                printf("NO\n");
            }
        }
    }
    return 0;
}

File: python/1003.py

n=input()
i=1
while i<=n:
	index=0
	count_P=0
	count_T=0
	sign_P=0
	sign_T=0
	sign=0
	sign_end=0
	string=raw_input()
	for j in string:
		index+=1
		if j!='P' and j!='A' and j!='T':
			print "NO"
			sign_end=1
			break

		if j=='P':
			sign_P=index
			count_P+=1
			sign=1

		if j=='T':
			if sign==0:
				print "NO"
				sign_end=1
				break
			sign_T=index
			count_T+=1
			if sign_T-sign_P==1:
				print "NO"
				sign_end=1
				break
	if sign_end==0:
		a=sign_P-1
		num_a=sign_T-sign_P-2
		if count_P==1 and count_T ==1:
			if (num_a+1)*a<=(index-sign_T):
				print "YES"
			else:
				print "NO"
		else:
			print "NO"
	i+=1

